File: tests/fixtures/java/Simple.java
package com.example;

import java.util.List;
import java.util.ArrayList;

/**
 * Greetable interface for objects that can produce greetings.
 */
interface Greetable {
    String greet(String name);
}

/**
 * Simple class demonstrating Java parser extraction.
 */
public class Simple implements Greetable {

    private final String prefix;

    public Simple(String prefix) {
        this.prefix = prefix;
    }

    @Override
    public String greet(String name) {
        return prefix + ", " + name + "!";
    }

    public List<String> greetAll(List<String> names) {
        List<String> results = new ArrayList<>();
        for (String name : names) {
            results.add(greet(name));
        }
        return results;
    }

    public static void main(String[] args) {
        Simple s = new Simple("Hello");
        System.out.println(s.greet("World"));
    }
}

File: tests/fixtures/go/simple.go
package main

import (
	"fmt"
	"strings"
)

// Greeter is a type that knows how to greet.
type Greeter struct {
	Name string
}

// String returns the string representation of a Greeter.
func (g Greeter) String() string {
	return "Greeter(" + g.Name + ")"
}

// greet returns a greeting for the given name.
func greet(name string) string {
	return "Hello, " + strings.TrimSpace(name) + "!"
}

func main() {
	g := Greeter{Name: "World"}
	fmt.Println(greet(g.Name))
	fmt.Println(g.String())
}
